lib/engine/game/g_18_uruguay/step/farm.rb

```diff
@@ -0,0 +1,104 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/assign'
+
+module Engine
+  module Game
+    module G18Uruguay
+      module Step
+        class Farm2 < Engine::Step::Assign
+          def setup
+            @farm_id = nil
+          end
+
+          def goods
+            @game.abilities_ignore_owner(current_entity, :assign_hexes, time: 'or_start', strict_time: false)
+          end
+
+          def used_this_or?
+            goods
+          end
+
+          def blocking_for_farm?
+            return false unless @round.operating?
+
+            used_this_or?
+          end
+
+          def actions(entity)
+            return [] unless entity.minor?
+
+            %w[assign].freeze
+          end
+
+          def description
+            'Deliver cattle'
+          end
+
+          def active?
+            blocking_for_farm?
+          end
+
+          def blocks?
+            active?
+          end
+
+          def neighbor_to_choosen_farm?(farm_id, hex_id)
+            @game.hex_by_id(farm_id).neighbors.find do |neighbor|
+              neighbor[1].id == hex_id \
+              && neighbor[1].tile.city_towns.size.positive?
+            end
+          end
+
+          def available_hex(entity, hex)
```

Comment:
I'm not following the logic in this method. It might be clearer if it were split into two codepaths, one if `@farm_id` is nil, the other if a farm has already been selected.

Reply:
Updated